cmd/update-rules/main.go

```diff
@@ -121,9 +125,26 @@ func load(rulesFile string) (*config.RepositoryRules, error) {
 	return rules, nil
 }
 
-func UpdateRules(rules *config.RepositoryRules, branch, goVer string) {
+func UpdateRules(rules *config.RepositoryRules, branch, goVer string, deleteRule bool) {
 	// run the update per destination repo in the rules
 	for j, r := range rules.Rules {
+		var deletedBranch bool
+		// To Check and Remove the existing/deprecated branch
+		if deleteRule {
+			for i, br := range r.Branches {
+				if br.Name == branch {
+					glog.Infof("Remove rule %s for %s", branch, r.DestinationRepository)
+					r.Branches = append(r.Branches[:i], r.Branches[i+1:]...)
```

Comment:
The assignment should be done to  `rules.Rules[j].Branches`, else the `r.Branches` is a local variable that is being created in the loop. So the final array is not being modified.

Reply:
look for last line  final array is being modified here 
`rules.Rules[j] = r`

Reply:
This change is yet to be done.

Reply:
Do you know what is the idiomatic way to write this piece of code? Thinking out loud, which format should we use.